### PR TITLE
Revert broken sed --in-place command

### PR DESCRIPTION
**Pull request by maresb**
Partial fix for #1520

### CODE REVIEW

File: base-notebook/start.sh [hunk 1]
```diff
@@ -176,11 +176,18 @@ else
     if ! whoami &> /dev/null; then
         echo "There is no entry in /etc/passwd for our UID. Attempting to fix..."
         if [[ -w /etc/passwd ]]; then
+
```
Comment (mathbunnyru) [inline base-notebook/start.sh:179]:
```suggestion
```

**Comment by maresb** [comment]
@mathbunnyru I'm usually not so aggressive about force-pushing, but I just can't bring myself to make a commit to remove a blank line. :)

**Comment by mathbunnyru** [comment]
That's fine :)

**Comment by maresb** [comment]
We also have the extraneous warnings from #1520... I'm wondering what's the best way to fix those.

File: base-notebook/start.sh [hunk 2]
```diff
@@ -177,10 +177,16 @@ else
...
-            echo "jovyan:x:$(id -u):$(id -g):,,,:/home/jovyan:/bin/bash" >> /etc/passwd
+            # We cannot use "sed --in-place" since sed tries to create a temp file in
+            # /etc/ and we may not have write access. Apply sed on our own temp file:
+            sed --expression="s/^jovyan:/nayvoj:/" /etc/passwd > /tmp/passwd
+            echo "jovyan:x:$(id -u):$(id -g):,,,:/home/jovyan:/bin/bash" >> /tmp/passwd
+            cat /tmp/passwd > /etc/passwd
+            rm /tmp/passwd
+
             echo "Added new jovyan user ($(id -u):$(id -g)). Fixed UID!"
+
```
Comment (mathbunnyru) [inline base-notebook/start.sh:189]: About blank lines - could you please remove this one as well? :)

**Review by mathbunnyru — APPROVED** [review]

File: base-notebook/start.sh [hunk 3]
```diff
@@ -177,10 +177,16 @@ else
...
         if [[ -w /etc/passwd ]]; then
             echo "Renaming old jovyan user to nayvoj ($(id -u jovyan):$(id -g jovyan))"
-            sed --in-place "s/^jovyan:/nayvoj:/" /etc/passwd
 
-            echo "jovyan:x:$(id -u):$(id -g):,,,:/home/jovyan:/bin/bash" >> /etc/passwd
+            # We cannot use "sed --in-place" since sed tries to create a temp file in
+            # /etc/ and we may not have write access. Apply sed on our own temp file:
+            sed --expression="s/^jovyan:/nayvoj:/" /etc/passwd > /tmp/passwd
+            echo "jovyan:x:$(id -u):$(id -g):,,,:/home/jovyan:/bin/bash" >> /tmp/passwd
+            cat /tmp/passwd > /etc/passwd
```
[the remark this answers is not in the document]
Comment (maresb) [reply base-notebook/start.sh:185]: Ya... so `/etc/passwd` has `g+w` and `jovyan` is in the `root` group.

File: base-notebook/start.sh (hunk 2 above)
Comment (maresb) [reply base-notebook/start.sh]: :)

**Comment by maresb** [comment]
I think we pulled the trigger too early on that PR. Should have written tests first. Oh well.

I think we should leave the false warnings in there for now. I don't have time to deal with them right away. I find the code before the PR more logical in this part.

**Comment by mathbunnyru** [comment]
@maresb yeah, our testing is not the best, there are too many cases and shell scripts are not tested easily.

So, what do you propose?
I can merge this one quickly, you checked, that this fixes the issue, right?
I would appreciate if you added a test for this later.

**Comment by maresb** [comment]
Ya, let's do a quick merge here. We'll work on test cases and warnings later.

**Comment by mathbunnyru** [comment]
Done

**Comment by consideRatio** [comment]
:heart: :tada: thanks for your work on making this land properly @maresb @mathbunnyru!!

**Pull request merged**